cocos/platform/android/CCGLViewImpl-android.cpp

```diff
@@ -127,6 +130,62 @@ void GLViewImpl::setIMEKeyboardState(bool bOpen)
     }
 }
 
+Rect GLViewImpl::getSafeAreaRect() const {
+    Rect safeAreaRect = GLView::getSafeAreaRect();
+    float deviceAspectRatio = 0;
+    if(safeAreaRect.size.height > safeAreaRect.size.width) {
+        deviceAspectRatio = safeAreaRect.size.height / safeAreaRect.size.width;
+    } else {
+        deviceAspectRatio = safeAreaRect.size.width / safeAreaRect.size.height;
+    }
+
+    float margin = DEFAULT_MARGIN_ANDROID / _scaleX;
+
+    bool isScreenRound = JniHelper::callStaticBooleanMethod("org/cocos2dx/lib/Cocos2dxHelper", "isScreenRound");
+    bool hasSoftKeys = JniHelper::callStaticBooleanMethod("org/cocos2dx/lib/Cocos2dxHelper", "hasSoftKeys");
+    if(isScreenRound) {
+        // edge screen (ex. Samsung Galaxy s7, s9, s9+, Note 9, Nokia 8 Sirocco, Sony Xperia XZ3, Oppo Find X...)
+        if(safeAreaRect.size.width < safeAreaRect.size.height) {
+            safeAreaRect.origin.y += margin * 2.f;
```

Comment:
when the screen is round, those 4 corners is round. so I think this line `origin.y += margin * 2.f;` should be changed to `origin.y += margin;`, just like landscape situation. is right?

Reply:
Unfortunately but no, this pice of code looks different type of screen ("edge screen" - https://9to5google.com/2016/05/06/how-to-edge-screen-galaxy-s7-edge/) - it is straight on one side and phisicly rounded around the vertical edges. I've tested on several devices with physical round corners but variable  `isScreenRound` doesn't true and it is true only if "edge" screen.

Reply:
I admit that this code will be changed and it will be necessary to introduce exceptions, for example there are already several devices that are rounded and at the edges and at the corners.

Reply:
thanks for explaining